src/main/webapp/js/script.js

```diff
@@ -0,0 +1,381 @@
+/**
```

Comment:
We can create a clean up PR after this. I think it is better to separate javascript util and UI related functions.  It could make code cleaner and avoid unnecessary merge conflicts.